### PR TITLE
Fix inability to join a `knock` room via space hierarchy view

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/element-hq/element-web/issues/27025

### CODE REVIEW

File: src/components/structures/SpaceHierarchy.tsx [hunk 1]
```diff
@@ -157,7 +158,8 @@ const Tile: React.FC<ITileProps> = ({
                 <Spinner w={24} h={24} />
             </AccessibleTooltipButton>
         );
-    } else if (joinedRoom) {
+    } else if (joinedRoom || room.join_rule === JoinRule.Knock) {
+        // We use the view flow for knock rooms as it allows us to reuse the request to join UX in RoomView
```
Comment (richvdh) [inline src/components/structures/SpaceHierarchy.tsx:162]: I don't really understand this comment. In particular:
 * what's a view flow?
 * what's the request we are reusing?

I think I get it after about the 10th reading: is this accurate?

```suggestion
        // If the room is knockable, show the "View" button even if we are not a member; that
        // allows us to reuse the "request to join" UX in RoomView.
```

**Review by richvdh — COMMENTED** [review]
Needs tests?

File: src/components/structures/SpaceHierarchy.tsx (hunk 1 above)
Comment (t3chguy) [reply src/components/structures/SpaceHierarchy.tsx:162]: Yes that is better English indeed

**Comment by t3chguy** [comment]
> Needs tests?

Knocking is in labs so technically it does not need tests until it leaves labs

**Review by richvdh — CHANGES_REQUESTED** [review]
> Knocking is in labs so technically it does not need tests until it leaves labs

As discussed elsewhere, I'd really like to see *some* sort of test for this anyway.

**Review by richvdh — APPROVED** [review]
lgtm. thanks for the test!

**Pull request merged**